.circleci/scripts/release-create-gh-release

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+set -u
+set -o pipefail
+
+if [[ "${CI:-}" != 'true' ]]
+then
+    printf '%s\n' 'CI environment variable must be set to true'
+    exit 1
+fi
+
+if [[ "${CIRCLECI:-}" != 'true' ]]
+then
+    printf '%s\n' 'CIRCLECI environment variable must be set to true'
+    exit 1
+fi
+
+printf '%s\n' 'Installing hub CLI'
+pushd "$(mktemp -d)"
+    curl -sSL 'https://github.com/github/hub/releases/download/v2.11.2/hub-linux-amd64-2.11.2.tgz' | tar xz
+    PATH="$PATH:$PWD/hub-linux-amd64-2.11.2/bin"
+popd
+
+current_commit_msg=$(git show -s --format='%s' HEAD)
```

Comment:
Does this amount to the changelog?

Our current changelog has a weird bug where it creates a number of "changelog-description" entries, and I usually end up manually removing those. We might want to also address that too, to achieve full (clean) automation, since having a good changelog is really nice.

Reply:
This is looking at specifically the version commit made on line 30 of the `release-bump-manifest-version` script above, not any of the commits in between this version and the last.

I took a look at the changelog script, and I think I know what's causing the problem (it's splitting on newlines in the commit body, treating each line as a separate commit). I can put up a separate PR to fix that, and maybe add some error handling.

Once that's fixed, perhaps the changelog can be updated as part of the `release-bump-manifest-version` script.

Reply:
I've made these changes here: https://github.com/MetaMask/metamask-extension/pull/6753

It's still not at the point where I'd trust it enough to include it in `release-bump-manifest-version` unfortunately. The commit messages are too inconsistent. We could certainly improve that, but that would likely introduce additional friction for contributors, which I'd rather avoid.

I might have an idea for how to accomplish this without adding much friction. We could use the squash-and-merge strategy for all PRs (e.g. disable the other two options). That'd ensure that the commit messages at least have _roughly_ the same format, at least as long as people stick with the defaults populated by GitHub (which adds `(#<PR>)` to the end of the commit subject).

We could take it a step further by adding a GitHub App to enforce certain things about the commit message, and make helpful suggestions. e.g. a commit linter, but only for the final squashed commit, not for the individual feature branch commits. I've seen something like that before - there might be an existing app that does this already, or at least approximates it. That also makes it easier to adopt additional conventions, like categorizing each commit. That would let us generate nicer changelogs.

Probably worth having a discussion about at some point. I'll have a quick look around to see how the tools in this space look - I haven't looked into this in a year or so.

Reply:
After further investigation, it looks like a GitHub App can't interact with the Squash & Merge commit message. There's no way to that commit *after* it was authored but *before* the merge - those happen at the same time right now.